### PR TITLE
feat: authentication endpoints

### PR DESCRIPTION
**Pull request by ElegantGracie**
Added signup, login and request for forgotten password endpoints

### CODE REVIEW

File: lib/controllers/auth-controller.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+        await user.save();
+
+    } catch (error) {
+        console.error(error);
+        return res.status(INTERNAL_SERVER_ERROR).send({message: "Error while saving user"})
+    }
+
+    const accessToken = generateToken(value.email, user, true);
+
```
Comment (Barkty) [inline lib/controllers/auth-controller.js:55]: @ElegantGracie how would the FE get access to the access tokens if you send via the headers? Do they have to the res.headers?

File: lib/controllers/auth-controller.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    return res.status(OK).header("Authorization", `Bearer ${accessToken}`).send({message: "User registered successfully", token: `Bearer ${accessToken}`});
+}
+exports.login = async (req, res) => {
+    const passwordPattern = /^(?=.*[a-zA-Z])(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{8,}$/;
+
+    const schema = Joi.object({
```
Comment (Barkty) [inline lib/controllers/auth-controller.js]: You can move the Joi validator to another file... Makes everywhere cleaner

File: lib/models/user-model.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const userSchema = new Schema(
+    {
```
Comment (Barkty) [inline lib/models/user-model.js:5]: You don't need to necessarily add _id in the schema.. that is auto-generated by mongodb

File: lib/utils/auth-utils.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+exports.getTransport = () =>
+  nodeMailer.createTransport({
+    host: process.env.EMAIL_HOST,
+    port: process.env.EMAIL_PORT,
+    auth: {
+      user: process.env.EMAIL_ADDRESS,
+      pass: process.env.EMAIL_PASSWORD
+    },
+  });
+
```
Comment (Barkty) [inline lib/utils/auth-utils.js:78]: You can create a js file that has your email tamplate

**Review by Barkty — CHANGES_REQUESTED** [review]
I made some comments in the changes... WE could have a call and see what needs to be fixed

File: lib/controllers/auth-controller.js (hunk 1 above)
Comment (ElegantGracie) [reply lib/controllers/auth-controller.js:55]: I have removed all response headers and left only the response body in my new commit.

File: lib/controllers/auth-controller.js (hunk 2 above)
Comment (ElegantGracie) [reply lib/controllers/auth-controller.js]: I have created a validator file.

File: lib/models/user-model.js (hunk 3 above)
Comment (ElegantGracie) [reply lib/models/user-model.js:5]: I wanted to use a readable _id but it's okay. I am going to use the autogenerated one.

**Comment by ElegantGracie** [comment]
I made a new commit. You can go through it to confirm I made the review changes.

**Pull request merged**